src/backend/distributed/utils/foreign_key_relationship.c

```diff
@@ -98,6 +104,108 @@ static void GetConnectedListHelper(ForeignConstraintRelationshipNode *node,
 static List * GetForeignConstraintRelationshipHelper(Oid relationId, bool isReferencing);
 
 
+/*
+ * GetForeignKeyConnectedRelationIdList returns a list of relation id's for
+ * relations that are connected to relation with relationId via a foreign
+ * key graph.
+ */
+List *
+GetForeignKeyConnectedRelationIdList(Oid relationId)
+{
+	/* use ShareRowExclusiveLock to prevent concurent foreign key creation */
+	LOCKMODE lockMode = ShareRowExclusiveLock;
+	Relation relation = try_relation_open(relationId, lockMode);
+	if (!RelationIsValid(relation))
+	{
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
```

Comment:
should add a test case for that

Reply:
08f57d6

Reply:
It seems that we can't test this as caller udf already checks this condition implicitly, as in 08f57d6.
However, it still makes sense to keep this check here as we will directly use this C api when implementing cascading logic in `create_citus_local_table` & `undistribute_table` udf's